images_to_samples.py

```diff
@@ -309,72 +316,109 @@ def main(params):
+    metadata = None
+    targ_ids = get_key_def('target_ids', params['sample'], None, expected_type=List)
+    class_prop = get_key_def('class_proportion', params['sample']['sampling_method'], None, expected_type=dict)
+    mask_reference = get_key_def('mask_reference', params['sample'], default=False, expected_type=bool)
 
-    experiment_name = get_key_def('mlflow_experiment_name', params['global'], default='gdl-training')
-    sample_path_name = (f'samples{samples_size}_overlap{overlap}_min-annot{min_annot_perc}_'
-                        f'{num_bands}bands_{experiment_name}')
+    # add git hash from current commit to parameters if available. Parameters will be saved to hdf5s
+    params['global']['git_hash'] = get_git_hash()
 
     # AWS
+    final_samples_folder = None
+    bucket_name = get_key_def('bucket_name', params['global'])
+    bucket_file_cache = []
     if bucket_name:
         s3 = boto3.resource('s3')
         bucket = s3.Bucket(bucket_name)
         bucket.download_file(csv_file, 'samples_prep.csv')
         list_data_prep = read_csv('samples_prep.csv')
-        if data_path:
-            final_samples_folder = data_path.joinpath("samples")
-        else:
-            final_samples_folder = "samples"
-        samples_folder = sample_path_name
-
     else:
         list_data_prep = read_csv(csv_file)
-        samples_folder = data_path.joinpath(sample_path_name)
 
-    if samples_folder.is_dir():
+    smpls_dir = data_path.joinpath(samples_folder_name)
+    if smpls_dir.is_dir():
         if debug:
             # Move existing data folder with a random suffix.
-            shutil.move(samples_folder, f'{str(samples_folder)}_{uuid.uuid4().hex[0:6]}')
+            last_mod_time_suffix = datetime.fromtimestamp(smpls_dir.stat().st_mtime).strftime('%Y%m%d-%H%M%S')
+            shutil.move(smpls_dir, data_path.joinpath(f'{str(smpls_dir)}_{last_mod_time_suffix}'))
         else:
-            raise FileExistsError(f'Data path exists: {samples_folder}. Remove it or use a different experiment_name.')
-    else:
-        tqdm.write(f'Writing samples to {samples_folder}')
-    Path.mkdir(samples_folder, exist_ok=False)  # TODO: what if we want to append samples to existing hdf5?
-    tqdm.write(f'Samples will be written to {samples_folder}\n\n')
+            raise FileExistsError(f'Data path exists: {smpls_dir}. Remove it or use a different experiment_name.')
+    Path.mkdir(smpls_dir, exist_ok=False)  # TODO: what if we want to append samples to existing hdf5?
 
-    tqdm.write(f'\nSuccessfully read csv file: {Path(csv_file).stem}\n'
-               f'Number of rows: {len(list_data_prep)}\n'
-               f'Copying first entry:\n{list_data_prep[0]}\n')
-    ignore_index = get_key_def('ignore_index', params['training'], -1)
-    meta_map, metadata = get_key_def("meta_map", params["global"], {}), None
+    import logging.config  # See: https://docs.python.org/2.4/lib/logging-config-fileformat.html
```

Comment:
hmmn, ansy when i see python 2, we are at python 3.x for GDL. it could be the functionality is the same in python3?

Reply:
I guess... I tested and it works well...